### PR TITLE
Feat/update rpc v0.1.0

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [x] The formatter, linter, and tests all run without an error
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

It updates RPC to the newest version

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

Yes, in terms of RPC.

### CODE REVIEW

File: starknet_py/net/rpc_schemas/rpc_schemas.py
Comment (cptartur) [inline starknet_py/net/rpc_schemas/rpc_schemas.py:145]: I wonder if accepting `None` as class_hash is a good idea. This seems like the most important field in the `DeployTransaction`.

Comment (cptartur) [reply starknet_py/net/rpc_schemas/rpc_schemas.py:145]: Perhaps we should get rid of `load_default` here entirely.

File: starknet_py/net/rpc_schemas/rpc_schemas.py
Comment (cptartur) [inline starknet_py/net/rpc_schemas/rpc_schemas.py:233]: Why do we need to construct this dict, couldn't we pass data directly to the schema?

File: starknet_py/net/rpc_schemas/rpc_schemas.py
Comment (cptartur) [inline starknet_py/net/rpc_schemas/rpc_schemas.py:224]: I wonder if it wouldn't make more sense to declare a separate schema for StateDiff and then have `state_diff = fields.Nested(StateDiffSchema(), ...)`. This way we'd only have to flatten this data in `make_dataclass`, without the need for all of this processing that is error prone.

File: starknet_py/net/rpc_schemas/rpc_schemas.py
Comment (cptartur) [inline starknet_py/net/rpc_schemas/rpc_schemas.py:312]: Are we testing these? Perhaps if devnet doesn't support pending transactions, we should mock one based off the RPC response and add tests for these.

File: starknet_py/tests/e2e/account/account_client_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/account/account_client_test.py:196]: 👁️

File: starknet_py/tests/e2e/client/client_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/client/client_test.py:267]: Would it be possible to replace this hardcoded value with a fixture?

File: starknet_py/tests/e2e/client/full_node_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/client/full_node_test.py:51]: Why was this test removed?

**Review by cptartur — COMMENTED** [review]
Left some changes.

Unrelated, but I wonder if we should be adding docstrings to some of our fixtures. Our tests are becoming very dependent on these.

File: starknet_py/tests/e2e/client/full_node_test.py
Comment (war-in) [reply starknet_py/tests/e2e/client/full_node_test.py:51]: Because get_block will not throw. We have default block_id="pending"

File: starknet_py/contract.py
Comment (cptartur) [inline starknet_py/contract.py:237]: Is it though? It doesn't seem like it from the function declaration?

File: starknet_py/tests/e2e/account/account_client_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/account/account_client_test.py:101]: This could use parametrized fixtures with `indirect=True` 😉

File: starknet_py/tests/e2e/client/full_node_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/client/full_node_test.py:131]:
```suggestion
        assert pending_transactions[0].hash == 0x1
```
Doesn't change anything in logic but i think it's more explicit since in mock you're using felts not ints.

File: starknet_py/contract.py
Comment (war-in) [reply starknet_py/contract.py:237]: All methods which use block_hash or block_number and FullNode/Gateway Client has the block_number parameter default to "pending"

Comment (cptartur) [reply starknet_py/contract.py:237]: Just to be safe in case we change anything in Client I think we could also include default parameter in `Contract` anyways.

File: starknet_py/tests/e2e/client/client_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/client/client_test.py:298]: Couldn't these also be replaced with the fixture you defined?

File: starknet_py/net/full_node_client.py
Comment (war-in) [reply starknet_py/net/full_node_client.py:195]: this will be used in the next 0.10.0 cairo

**Review by Solpatium — APPROVED** [review]
Looks ok for me, but @cptartur has more info about RPC, so please wait for his approval before merging.

File: starknet_py/tests/e2e/client/client_test.py
Comment (cptartur) [inline starknet_py/tests/e2e/client/client_test.py:344]: I'd use a fixture here if possible.

**Review by cptartur — COMMENTED** [review]
Except for one comment, looks fine to me 👍

**Pull request merged**
